great_expectations/expectations/expectation.py

```diff
@@ -3005,26 +3005,25 @@ def _format_map_output(
                 {"error": "partial_exception_counts requires a hashable type"}
             ]
         finally:
+            if unexpected_index_list is not None:
+                return_obj["result"].update(
+                    {
+                        "partial_unexpected_index_list": unexpected_index_list[
+                            : result_format["partial_unexpected_count"]
+                        ],
+                    }
+                )
             return_obj["result"].update(
-                {
-                    "partial_unexpected_index_list": unexpected_index_list[
-                        : result_format["partial_unexpected_count"]
-                    ]
-                    if unexpected_index_list is not None
-                    else None,
-                    "partial_unexpected_counts": partial_unexpected_counts,
```

Comment:
This if statement was making it so that `partial_unexpected_index_list` was always being included in the `return_obj` even if it was invalid (ie not supported by the `ExecutionEngine`)